File: python/sources/treina3.py
import random
import os

imagens = 120
amostra = [0] * imagens

for i in range(imagens):
    with open(f"imagens/amostra{i:04d}.bmp", "rb") as f:
        buf = f.read()
    amostra[i] = sum(buf[54:])

bias = random.random()  # entre 0 e 1

for x in range(300000): # ajusta o bias milhares de vezes
    alguma = random.randint(0, imagens - 1)  # uma imagem qualquer

    if amostra[alguma] > bias:
        bias += bias * 0.0001 # um pouco a mais
    elif amostra[alguma] < bias:
        bias -= bias * 0.0001 # um pouco a menos

    print(f"finalBias: {bias}")

with open('bias2.txt', 'w', encoding='utf-8') as f:
    f.write(f"{bias}\n") # salva o bias final no arquivo

File: nodeJs/sources/treina3.js
fs=require('fs');

imagens=120; //qtd de imagens para treinar
amostra=new Array(imagens).fill(0);

bias=Math.random(); //inicializa bias aleatório

for(i=0;i<imagens;i++){
	buf=fs.readFileSync(`imagens/amostra${i.toString().padStart(4, '0')}.bmp`);
	soma=0;
	for(j=54;j<buf.length;j++)soma += buf[j];
	amostra[i]=soma; //gera um 'weight' que representa a imagem
}

for(x=0;x<200000;x++){ //ajusta o bias centenas de vezes
	alguma=parseInt(Math.random()*imagens); //uma imagem qualquer

	//ajusta o bias pouco a pouco
	if(amostra[alguma]>bias)bias+=bias*0.00008; //um pouco a mais
	if(amostra[alguma]<bias)bias-=bias*0.00008; //um pouco a menos

	console.log(`biasfinal = ${bias.toFixed(6)}`);
}
//salva o bias no arquivo
fs.writeFileSync('bias2.txt',`${bias}\n`,'utf-8');

console.log("\nTreinamento finalizado. Pesos salvos em 'bias2.txt'")
